app/src/main/java/com/fastlink/zekrapp/presentation/ZekirScreen.kt

```diff
@@ -0,0 +1,198 @@
+    val category = viewModel.getCategoryById(categoryId)
+    val zekirs = viewModel.getZekirByCategoryId(category.id)
+    val zekirNumber = remember {
+        mutableIntStateOf(0)
+    }
+    val zekirCounter = remember {
+        mutableIntStateOf(0)
+    }
+    LaunchedEffect(true) {
+        toast(" الذكر ${zekirNumber.intValue + 1} من ${zekirs.size}")
+    }
+    val animatedBorder by animateFloatAsState(
+        targetValue =
+        (zekirCounter.intValue.toFloat() / zekirs[zekirNumber.intValue].counterNumber.toFloat()) * 360f,
+        label = "",
+        animationSpec = tween(1000)
+    )
+
+    fun onClicked() {
+        if (zekirCounter.intValue < zekirs[zekirNumber.intValue].counterNumber) {
+            zekirCounter.intValue++
+            if (zekirCounter.intValue == zekirs[zekirNumber.intValue].counterNumber) {
+                scope.launch {
+                    delay(1500)
+                    if (zekirNumber.intValue < zekirs.size - 1) {
+                        zekirCounter.intValue = 0
+                        zekirNumber.intValue++
+                        scope.launch {
+                            toast(" الذكر ${zekirNumber.intValue + 1} من ${zekirs.size}")
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    Scaffold(
+        topBar = {
+            ZekirScreenAppBar(title = category.name)
+        },
+        floatingActionButton = {
+            FloatingActionButton(
+                onClick = {
+                    onClicked()
+                },
+                elevation = FloatingActionButtonDefaults.elevation(
+                    defaultElevation = 10.dp,
+                    pressedElevation = 10.dp,
+                ),
+                backgroundColor = MaterialTheme.colorScheme.primaryContainer,
+                modifier = Modifier
+                    .size(75.dp)
+                    .clip(CircleShape)
+                    .background(MaterialTheme.colorScheme.onPrimary)
+                    .archBorder(animatedBorder = animatedBorder),
+            ) {
+                Text(
+                    text = zekirCounter.intValue.toString(),
+                    style = MaterialTheme.typography.titleLarge,
+                    color = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+            }
+        },
+
+
+        isFloatingActionButtonDocked = true,
+        floatingActionButtonPosition = FabPosition.Center,
+        bottomBar = {
+            ZekirScreenBottomAppBar(category = category,
+                onCopyIconClicked = { clipboardManager.setText(AnnotatedString(zekirs[zekirNumber.intValue].description)) }
+            )
+        }
+    ) {
+        Column(
+            horizontalAlignment = Alignment.End,
+            modifier = Modifier
+                .padding(it)
+                .fillMaxSize()
+                .verticalScroll(
+                    rememberScrollState()
+                )
+                .clickable(
```

Comment:
Same as Ahmed mentioned. You should try to make the ui part only for showing, and leave the logic for the view model.